pyqtgraph/graphicsItems/PlotDataItem.py

```diff
@@ -38,6 +38,85 @@ class PlotDataItem(GraphicsObject):
+        def __init__(self, x, y):
+            super().__init__()
+            assert x is not None
+            assert y is not None
+            self.x = x
+            self.y = y
+            self._dataRect = None
+            self.containsNonfinite = None
+            
+        # def copy(self):
+        #     """ Provides a shallow copy of this PlotDataset """
+        #     dataset = self.PlotDataSet( self.x, self.y )
+        #     dataset._dataRect = self._dataRect
+        #     dataset.containsNonfinite = self.containsNonfinite
+        #     return dataset
+            
+        def _updateDataRect(self):
+            """ Find bounds of plotable data """
+            if self.y is None or self.x is None:
+                return None
+            selection = np.isfinite(self.y) # We are looking for the bounds of the plottable data set. Infinite and Nan are ignored. 
+            all_y_are_finite = selection.all() # False if all values are finite, True if there are any non-finites
+            try:
+                ymin = np.min( self.y[selection] ) # find minimum of all finite values
+                ymax = np.max( self.y[selection] ) # find maximum of all finite values
+            except ValueError: # is raised then there are no finite values
+                ymin = np.nan
+                ymax = np.nan
+                
+            selection = np.isfinite(self.x) # We are looking for the bounds of the plottable data set. Infinite and Nan are ignored. 
+            all_x_are_finite = selection.all() # False if all values are finite, True if there are any non-finites
+            try:
+                xmin = np.min( self.x[selection] ) # find minimum of all finite values
+                xmax = np.max( self.x[selection] ) # find maximum of all finite values
+            except ValueError: # is reaised then there are no finite values
+                xmin = np.nan
+                xmax = np.nan
+                
+            self._dataRect = QtCore.QRectF( QtCore.QPointF(xmin,ymin), QtCore.QPointF(xmax,ymax) )
+            self.containsNonfinite = not (all_x_are_finite and all_y_are_finite) # This always yields a definite True/False answer
+
+        def dataRect(self):
+            """
+            Returns a bounding rectangle (as QRectF) for the finite subset of data.
+            If there is an active mapping function, such as logarithmic scaling, then bounds represent the mapped data. 
+            Will return None if there is no data or if all values (x or y) are NaN.
+            """
+            if self._dataRect is None: 
+                self._updateDataRect()
+            return self._dataRect
+            
+        def applyLogMapping(self, logMode):
+            """
+            Applies log10 as a mapping transformation according to logmode.
+            logmode is a two-element list of Booleans requesting log mode for x and y axis.
+            """
+            all_x_finite = False
+            if logMode[0]:
```

Comment:
wonder if we should move this if block into it's own "private" function, since the code is just repeated right below.

Reply:
Oops. This moved to PlotDataset rather late, and this copy should no longer be needed.